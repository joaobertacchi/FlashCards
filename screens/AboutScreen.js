// @flow

import React from 'react';
import {
  Image, ScrollView, StyleSheet, Text, View,
} from 'react-native';
import { WebBrowser, Updates } from 'expo';
import { connect } from 'react-redux';

import Button from '../components/Button';

import { handleResetDecks } from '../actions/decks';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  developmentModeText: {
    marginBottom: 20,
    color: 'rgba(0,0,0,0.4)',
    fontSize: 14,
    lineHeight: 19,
    textAlign: 'center',
  },
  contentContainer: {
    paddingTop: 10,
  },
  welcomeContainer: {
    alignItems: 'center',
    marginTop: 10,
    marginBottom: 10,
  },
  welcomeImage: {
    width: 200,
    height: 100,
    resizeMode: 'contain',
  },
  getStartedContainer: {
    alignItems: 'center',
    marginHorizontal: 50,
  },
  linkText: {
    fontSize: 14,
    color: '#2e78b7',
  },
});

class AboutScreen extends React.Component<*, *> {
  static navigationOptions = {
    title: 'About Me',
  };

  handleResetStorage = () => {
    const { resetStorage } = this.props;
    resetStorage();
    Updates.reloadFromCache();
  };

  handleMorePress = () => {
    WebBrowser.openBrowserAsync('https://www.linkedin.com/in/jebertacchi/');
  };

  render() {
    return (
      <ScrollView style={styles.container} contentContainerStyle={styles.contentContainer}>
        <View style={styles.welcomeContainer}>
          <Button onPress={this.handleResetStorage} text="Reset storage" type="danger" />
        </View>
        <View style={styles.welcomeContainer}>
          <Image
            source={require('../assets/images/joao-bertacchi.jpg')}
            style={styles.welcomeImage}
          />
        </View>
        <View style={styles.getStartedContainer}>
          <Text style={styles.developmentModeText}>
            Currently working as Software Developer Engineer for ProFUSION Embedded Systems. Worked
            for 10+ years as software development engineer using primarily C, C++, Python, Ruby, and
            Java languages. Experience with Scrum and CCPM project management. Co-founder of
            AulaViva Tecnologia Educacional, an EdTech company where I work as CTO doing product
            management and web development.
            {' '}
            <Text onPress={this.handleMorePress} style={styles.linkText}>
              More...
            </Text>
          </Text>
        </View>
      </ScrollView>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  resetStorage: () => dispatch(handleResetDecks()),
});

export default connect(
  null,
  mapDispatchToProps,
)(AboutScreen);
