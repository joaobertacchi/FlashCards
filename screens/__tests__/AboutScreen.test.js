import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import AboutScreen from '../AboutScreen';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};

describe('AboutScreen snapshot', async () => {
  it('renders as expected', () => {
    const wrapper = shallow(<AboutScreen />, { context: { store: mockStore(initialState) } });
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it.skip('click in reset storage', () => {
    AboutScreen.prototype.handleResetStorage = jest.fn();
    const wrapper = shallow(<AboutScreen />, { context: { store: mockStore(initialState) } });
    const render = wrapper.dive();
    render.find('Button').forEach((child) => {
      child.simulate('click');
    });
    expect(AboutScreen.prototype.handleResetStorage).toHaveBeenCalled();
  });
});
