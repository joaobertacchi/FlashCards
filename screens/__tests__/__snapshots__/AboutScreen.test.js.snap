// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen snapshot renders as expected 1`] = `
<ScrollViewMock
  contentContainerStyle={
    Object {
      "paddingTop": 10,
    }
  }
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <Button
      disabled={false}
      onPress={[Function]}
      text="Reset storage"
      type="danger"
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <Component
      source={1}
      style={
        Object {
          "height": 100,
          "resizeMode": "contain",
          "width": 200,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "marginHorizontal": 50,
      }
    }
  >
    <Component
      style={
        Object {
          "color": "rgba(0,0,0,0.4)",
          "fontSize": 14,
          "lineHeight": 19,
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      Currently working as Software Developer Engineer for ProFUSION Embedded Systems. Worked for 10+ years as software development engineer using primarily C, C++, Python, Ruby, and Java languages. Experience with Scrum and CCPM project management. Co-founder of AulaViva Tecnologia Educacional, an EdTech company where I work as CTO doing product management and web development.
       
      <Component
        onPress={[Function]}
        style={
          Object {
            "color": "#2e78b7",
            "fontSize": 14,
          }
        }
      >
        More...
      </Component>
    </Component>
  </Component>
</ScrollViewMock>
`;
